server/src/main/java/org/opensearch/indices/TermsLookup.java

```diff
@@ -84,12 +108,20 @@ public TermsLookup(StreamInput in) throws IOException {
             in.readOptionalString();
         }
         id = in.readString();
+        // if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+        // if (in.readBoolean()) {
+        // query = in.readOptionalWriteable(inStream -> inStream.readNamedWriteable(QueryBuilder.class));
+        // }
+        // }
         path = in.readString();
         index = in.readString();
         routing = in.readOptionalString();
         if (in.getVersion().onOrAfter(Version.V_2_17_0)) {
             store = in.readBoolean();
         }
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
The version guard needs to be "on or after" the current version, which 3.2. This is likely the cause of the various bwc test failures you're seeing.